persistence/src/main/java/cz/fi/muni/pa165/dao/VisitDaoImpl.java

```diff
@@ -115,8 +134,11 @@ public void delete(Visit visit) {
      * @return list of all visits for given date
      */
     public List<Visit> findByDate(Date date) {
-
-        throw new UnsupportedOperationException();
+        try {
+            return em.createQuery("select v from Visit v where date = :date",
+                    Visit.class).setParameter("date", date).getResultList();
+        } catch (NoResultException nrf) {
+            return null;
```

Comment:
what about the exact times?, maybe use day granularity, or redo it to search for ranges